From 118ff873a9a8214e8c1aed7c1a8f421ee72889a6 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Fri, 28 Apr 2023 14:30:37 -0400
Subject: [PATCH 22/35] Add gnu-stack notes

In binutils-2.39, GNU ld has added a warning when linking objects that
don't have a .note.GNU-stack section.  Since we build with
"--fatal-warnings", this causes a failure:

ld: warning: cert.o: missing .note.GNU-stack section implies executable stack
ld: NOTE: This behaviour is deprecated and will be removed in a future version of the linker

This patch adds those sections to all of the .S files, so that they will
be in the objects, as well as updating our gnu-efi tree the same way.

Signed-off-by: Peter Jones <pjones@redhat.com>
[AM: removed gnu-efi part]
Signed-off-by: Alexey Makhalov <amakhalov@vmware.com>
---
 cert.S     | 1 +
 sbat_var.S | 1 +
 2 files changed, 2 insertions(+), 0 deletion(-)

diff --git a/cert.S b/cert.S
index 3697033..9642efc 100644
--- a/cert.S
+++ b/cert.S
@@ -52,3 +52,4 @@ vendor_deauthorized:
 #endif
 .Lvendor_deauthorized_end:
 .Lcert_table_end:
+	.section .note.GNU-stack,"a"
diff --git a/sbat_var.S b/sbat_var.S
index 2a813a4..7854ade 100644
--- a/sbat_var.S
+++ b/sbat_var.S
@@ -20,3 +20,4 @@ sbat_var_payload_header:
 .Lsbat_var_latest:
 	.ascii SBAT_VAR_LATEST
 	.byte 0
+	.section .note.GNU-stack,"a"
-- 
2.39.0

